src/components/NewBlog/NewBlog.js

```diff
@@ -0,0 +1,100 @@
+import React, { useRef, useState } from 'react';
+import './NewBlog.css';
+import { useDispatch, useSelector } from 'react-redux';
+import { addNewBlog } from '../../store/blogSlice';
+import { LOCAL_TYPE, NEW_BLOG } from '../../constants/blog-constants';
+
+const NewBlogModal = ({ onClose }) => {
+  const dispatch = useDispatch();
+  const darkMode = useSelector((state) => state.theme.darkMode);
+  const [formData, setFormData] = useState({ title: '', details: '', type: LOCAL_TYPE, photo: '' });
+  const [selectedImage, setSelectedImage] = useState(null);
+  const fileInputRef = useRef(null);
+
+
+  /**
+   * Handle input change
+   * @param {*} e 
+   */
+  const handleInputChange = (e) => {
+    const { name, value } = e.target;
+    setFormData({ ...formData, [name]: value });
+  };
+
+  /**
+   * Handle choose image
+   */
+  const handleChooseFileClick = () => {
+    fileInputRef.current.click();
```

Comment:
What if current is null?

Reply:
I can add the optional notation to handle on null cases. In this case, we would click on the input that is currently displayed as none. I have made a change in the below code, to check if the file is not null before executing the file reader code.

const handleFileInputChange = (e) => {
    e.preventDefault();
    const file = e.target.files[0];
    if(file){
      const reader = new FileReader();
      reader.onload = (event) => {
        setFormData({ ...formData, photo: event.target.result });
        setSelectedImage(event.target.result);
      };
      reader.readAsDataURL(file);
    }
   
  };